### PR TITLE
Make scripts (mostly) POSIX compat; add shellcheck

### PR DESCRIPTION
**Pull request by mbargull**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

The activation scripts cannot be run in non-Bash-compatible shells, e.g., when run through `/bin/sh`/`/bin/dash` on Debian.
This is an attempt to allow support those shells.

I did not yet test these changes.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -14,6 +14,7 @@ requirements:
   build:
     # this dep is for us to be able to use gcc -dumpmachine to get the CBUILD value
     - gcc_impl_{{ target_platform }}
+    - shellcheck
```
Comment (mbargull) [inline recipe/meta.yaml:17]: Are there any downsides in adding additional build-time dependencies here?

File: recipe/activate-gcc.sh [hunk 2]
```diff
@@ -190,16 +193,16 @@ else
   fi
 
   # fix prompt for zsh
-  if [[ -n "${ZSH_NAME:-}" ]]; then
+  if [ -n "${ZSH_NAME:-}" ]; then
     autoload -Uz add-zsh-hook
 
     _conda_clang_precmd() {
-      HOST="${CONDA_BACKUP_HOST}"
+      export HOST="${CONDA_BACKUP_HOST}"
```
Comment (beckermr) [inline recipe/activate-gcc.sh:200]: IDK about this export. :/

File: recipe/activate-gcc.sh [hunk 3]
```diff
@@ -190,16 +193,16 @@ else
...
     }
     add-zsh-hook -Uz precmd _conda_clang_precmd
 
     _conda_clang_preexec() {
-      HOST="${CONDA_TOOLCHAIN_HOST}"
+      export HOST="${CONDA_TOOLCHAIN_HOST}"
```
Comment (beckermr) [inline recipe/activate-gcc.sh:205]: Same here.

File: recipe/activate-gcc.sh (hunk 2 above)
Comment (mbargull) [reply recipe/activate-gcc.sh]: Can you elaborate?
I just pushed a commit to not change it. But it shouldn't be important either way since `HOST` is already exported by the activation script.

Comment (beckermr) [reply recipe/activate-gcc.sh]: Yeah I am just worried about changing this stuff too much. These changes are hard to test in all of the edge cases and so things are somewhat brittle.

File: recipe/meta.yaml (hunk 1 above)
Comment (beckermr) [reply recipe/meta.yaml]: Not that I know of.
